src/App.jsx

```diff
@@ -28,7 +28,11 @@ function App() {
             </nav>
             <Routes>
               <Route path="/products" element={<Products />} />
-              <Route path="/orders" element={<Orders />} />
+              <Route path="/orders" element={
+                <PrivateRoute>
+                  <Orders />
+                </PrivateRoute>
+                } />
```

Comment:
We should use AuthContext for handling authentication, and private route can use a flag from that context, this will give you better handling of token and user info also it will avoid props drilling

Reply:
also to store token please don't use local storage, its not secure, go for http only Cookies